x-pack/plugins/license_management/public/store/reducers/licenseManagement.js

```diff
@@ -105,6 +100,12 @@ export const shouldShowStartTrial = state => {
     (licenseType !== 'platinum' || isExpired(state))
   );
 };
+
+export const shouldShowRequestTrialExtension = state => {
```

Comment:
This is probably outside the scope of this PR, but I was having a hard time tracing the relationships between these functions and seeing what kind of data they're interested in because they all accept a `state` object. I think we could make this code more transparent by changing them to accept more granular arguments, e.g. `license`, `licenseType`, `uploadStatus`.

Reply:
they are all selectors and thus operate on entire state object.

Reply:
You're right, though I still think there's plenty of room for improving the readability of this code while preserving its identify as a collection of selectors. For example, `getExpirationDate` and `isImminentExpiration` are only used internally, so they don't need to be coupled to the `state` object:

```js
export const getExpirationDate = license => {
  const { type, expiryDateInMillis } = license;
  //basic licenses do not expire
  if (type === 'basic') {
    return null;
  }

  if (expirationMillis) {
    return moment.tz(expirationMillis, moment.tz.guess());
  } else {
    return null;
  }
};

export const isImminentExpiration = license => {
  const now = new Date();
  const expirationDate = getExpirationDate(license);
  return (
    expirationDate &&
    expirationDate.isAfter(now) &&
    expirationDate.diff(now, 'days') <= WARNING_THRESHOLD_IN_DAYS
  );
};
```

Reply:
TypeScript will also help a lot here.

Reply:
Seems like the same effect could be achieved with decomposition in those methods:
`export const isImminentExpiration = ({ license })  => {` ...

At any rate, this is all beyond the scope of this simple logic shift PR I think.